WcaOnRails/app/helpers/persons_helper.rb

```diff
@@ -31,4 +31,72 @@ def return_podium_class(result)
       end
     end
   end
+
+  def delegate_badge(kind)
+    content_tag(:span, class: "badge delegate-badge") do
+      link_to(t("enums.user.delegate_status." + kind), "/delegates", title: t("enums.user.delegate_status." + kind), data: { toggle: "tooltip", placement: "bottom" })
+    end
+  end
+
+  def officer_badge(team)
+    content_tag(:span, class: "badge officer-badge") do
+      link_to(team, "/teams-committees#officers", title: t('about.structure.officers.name'), data: { toggle: "tooltip", placement: "bottom" })
+    end
+  end
+
+  def team_badge(team, kind, extra)
+    content_tag(:span, class: "badge team-" + kind + "-badge") do
+      link_to(team.acronym + extra, "/teams-committees#" + team.acronym, title: team.name, data: { toggle: "tooltip", placement: "bottom" })
+    end
+  end
+
+  def all_user_badges(user)
+    # Checks if person is any kind of Delegate, officer, Board member, any Team leader, any team sennior member,
+    # any team member and display all the badges in that order
+    badges = []
+    if user.any_kind_of_delegate? # Delegates
+      if user.senior_delegate?
+        badges.push(delegate_badge("senior_delegate"))
+      elsif user.full_delegate?
+        badges.push(delegate_badge("delegate"))
+      else
+        badges.push(delegate_badge("candidate_delegate"))
+      end
+    end
+
+    user.current_teams.select { |team| Team.all_officers.include? team }.map(&:name).each do |team| # Officers
+      badges.push(officer_badge(team))
+    end
+    if Team.wfc.current_members.select(&:team_leader).map(&:user).include?(user)
+      badges.push(officer_badge(t('about.structure.treasurer.name')))
+    end
+
+    if user.board_member? # Board
+      badges.push(team_badge(Team.board, "member", ""))
+    end
+
+    if user.leader_of_any_official_team? || user.team_leader?(Team.wac) # Team leaders
+      user.teams_where_is_leader.each do |team|
```

Comment:
I think for this, the senior badges, and the team membership the whole loop can be simplified into a one-liner:
`badges.push(user.teams_where_is_leader.map { |team| team_badge(team, "leader", " " + t('about.structure.leader')) })`.
For the "current_teams" loop you may want to include a select before the map:
`all_teams = Team.all_official + Team.all_councils` then
`user.current_teams.select { |t| all_teams.include?(t) }.reject { |t| user.team_leader?(t) || user.team_senior_member?(t) }.map....`

It may be worth to add a scope on teams like "official and councils", that we may apply on `current_teams` (so that the filtering is done by sql rather than by ruby).

Reply:
The 3 one-liners work like a charm :+1:  ( I splitted the 3rd one because it was a very long one-liner :stuck_out_tongue: )

about the other change, you mean adding
`  def self.all_official_and_councils
    self.all_official + self.all_councils
  end`
to team model?

Then I have doubts if just updating this 
`scope :in_official_team, -> { where(team_id: Team.all_official.map(&:id)) }`
on team_member model or adding a new line and then changing user model subsequently, I'll have to dig deeper into the differences of councils to see what fits better, for example, as of now a council team_leader is not an eligible voter which will change if I just change the above line (my guess is it should change and we haven't noticed it because the current leader already has voting rights as a Delegate) 

Anyway, I'll dig deeper unless someone solves my doubts on this.

Reply:
Looking at the code, if we add the proposed change (Adding all_official_and_councils and then pointing in_official_team to that would change the following:

- eligible voters and senior_leader_voters, adding WAC leader and senior members
- WAC leader would be added to leaders mailing list

If we just changed Team.all_official to add WAC, then we also would have to change this page: https://www.worldcubeassociation.org/contact/ (which I'm not sure it's linked anywhere).

I would add the first proposed change but not the second, because it also makes sense to have it there for future uses. If you agree, I'll commit the changes and squash if necessary.

Reply:
I've just gone ahead and added all_official_and councils without changing where in_official_team points because it's not needed for this PR, I still think this should be changed, though.

Reply:
Well the thing is that not all council members are WCA Staff, which is why councils have this special treatment.
I like how you've handled it though.